src/client/pages/Geo/GeoMap/hooks/useFetchGteTreeCoverPercentLayer.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react'
+
+import { LayerKey, LayerSectionKey } from '@meta/geo'
+
+import { useAppDispatch } from '@client/store'
+import { GeoActions, useGeoLayer } from '@client/store/ui/geo'
+import { useCountryIso } from '@client/hooks'
+
+export const useFetchGteTreeCoverPercentLayer = (sectionKey: LayerSectionKey, layerKey: LayerKey) => {
+  const dispatch = useAppDispatch()
+  const countryIso = useCountryIso()
+  const layerState = useGeoLayer(sectionKey, layerKey)
+  const gteTreeCoverPercent = layerState?.options?.gteTreeCoverPercent
+
+  useEffect(() => {
+    if (gteTreeCoverPercent === undefined) return // Skip when the property is not set
+    dispatch(GeoActions.postLayer({ countryIso, sectionKey, layerKey }))
+  }, [countryIso, layerKey, gteTreeCoverPercent, sectionKey, dispatch])
```

Comment:
This is the same than useFetch agreement level ... can't it be one and only one fetch ? Then based on the layer , you parse the params ?

Reply:
Wouldn't that make it harder to implement logic specific to those layers in the future? like caching (which would need to be handled differently for agreement).

Reply:
Also, we need to add logic to re-fetch the agreement layer when the selected layers change. My initial thought was to do that from `toggleLayer` in the slice, but that is not possible since that would be a side effect. I suppose then it would need to go in this hook, making it different from the `useFetchGte...`. Any other suggestions of where that logic could go?